src/operations/search/mod.rs

```diff
@@ -776,9 +776,26 @@ pub struct SearchHitsHitsResult<T> {
     #[serde(rename="_routing")]
     pub routing: Option<String>,
     pub fields: Option<Value>,
-    pub highlight: Option<HighlightResult>
+    pub highlight: Option<HighlightResult>,
+    pub inner_hits: Option<HashMap<String, Value>>
 }
 
+impl<T> SearchHitsHitsResult<T>
+    where T: DeserializeOwned {
+
+    pub fn inner_hit<S>(&self, key: &str) -> Option<SearchHitsResult<S>> where S: DeserializeOwned {
```

Comment:
This should perhaps be a `Result<Option<SearchHitsResult<S>>, EsError>` so we can tell apart the difference between a missing value and a malformed value?

`serde_json` errors will auto-promote to `EsError`s so replacing the `.ok()` on line 791 with a `?` would do the trick with the larger return type.

Reply:
I couldn't use `?` as I find the need to specify to rustc what kind of error are we dealing with - https://github.com/benashford/rs-es/pull/111/commits/aabd7077752f585ebaa65b483442456a03964d82